packages/compass-web/src/entrypoint.tsx

```diff
@@ -298,49 +298,38 @@ const CompassWeb = ({
             <WithAtlasProviders>
               <ConnectionStorageProvider value={connectionStorage.current}>
                 <ConnectionsManagerProvider value={connectionsManager.current}>
-                  <CompassInstanceStorePlugin>
-                    <FieldStorePlugin>
-                      {isConnected && connectionInfo ? (
-                        <AppRegistryProvider
-                          key={connectionInfo.id}
-                          scopeName="Connected Application"
-                        >
-                          <ConnectionInfoProvider
-                            connectionInfoId={connectionInfo.id}
+                  <CompassConnections
+                    onConnectionAttemptStarted={onConnectionAttemptStarted}
+                    onConnectionFailed={onConnectionFailed}
+                    onConnected={onConnected}
```

Comment:
In theory it would be good to also move the telemetry emitting inside compass-connections and remove this unneccessaty interface, on practice this would also require to refactor the whole test suite for those special telemetry events and I would prefer to do this work in a follow up as it would be hard to review otherwise.

This already fixes an issue of multiple connections not emitting telemetry events as the callbacks are now called without regarding which of the separate connection flows (multiple connections or single one) was triggered